libs/features/japanese-word-wrap.js

```diff
@@ -1,27 +1,85 @@
+/**
+ * Update the model to prevent line breaks from occurring for the specified word.
+ * @param {*} parser The BudouX parser to update.
+ * @param {string} pattern The pattern that should be updated in the model.
+ * @param {number} score The score that should be added from the pattern in the model.
+ * @param {string} [markerSymbol='#'] The marker symbol to look for in the pattern string.
+ * Defaults to '#'.
+ */
+function updateParserModel(parser, pattern, score, markerSymbol = '#') {
+  const markerPos = pattern.indexOf(markerSymbol);
+  if (markerPos === -1) {
+    console.warn('No marker symbol found in line break pattern string');
+    return;
+  }
+
+  if (markerPos !== pattern.lastIndexOf('#')) {
+    console.warn('Two or more marker symbols cannot be specified. Only the first marker is applied');
+  }
+
+  const former = pattern.slice(Math.max(markerPos - 3, 0), markerPos);
+  const latter = pattern.slice(markerPos + 1, Math.min(markerPos + 4, pattern.length));
+
+  if (former.length < 2 || latter.length < 2) {
+    console.warn('At least two characters must be specified before and after the marker symbol');
+    return;
+  }
+
+  if (former.length === 3) {
+    parser.model.set(`TW1:${former}`, score);
+  } else if (former.length === 2) {
+    parser.model.set(`BW1:${former}`, score);
+  }
+
+  if (latter.length === 3) {
+    parser.model.set(`TW4:${latter}`, score);
+  } else if (latter.length === 2) {
+    parser.model.set(`BW3:${latter}`, score);
+  }
+}
+
 /**
  * Searches for Japanese text in headings and applies a smart word-breaking algorithm by inserting
  *  <wbr> between semantic blocks. This allows browsers to break japanese sentences correctly.
- * @param {*} config The config.
- * @param {*} [area=document] The HTML element or Document.
- * @param {string} [selector='h1, h2, h3, h4, h5, h6'] The target selector you want to apply Japanese word wrap.
+ * @param {*} config The milo config.
+ * @param {*} options The options to control line breaks.
  */
-export async function controlLineBreaksJapanese(config, area = document, selector = 'h1, h2, h3, h4, h5, h6') {
+export default async function controlLineBreaksJapanese(config, options = {}) {
   const { miloLibs, codeRoot } = config;
+  const {
+    scopeArea = document,
+    budouxSelector = 'h1, h2, h3, h4, h5, h6',
+    budouxThres = 2000,
+    bwSelector = null,
+    lineBreakOkPatterns = [],
+    lineBreakNgPatterns = [],
+  } = options;
   const base = miloLibs || codeRoot;
 
   // The thresould value to control word break granularity for long semantic blocks.
-  const WORD_SEG_THRES = 8;
   const { loadDefaultJapaneseParser } = await import(`${base}/deps/budoux-index-ja.min.js`);
   const parser = loadDefaultJapaneseParser();
-  area.querySelectorAll(selector).forEach((el) => {
-    // apply budoux to headings
-    parser.applyElement(el, { wordSegThres: WORD_SEG_THRES });
+
+  // Update model
+  const SCORE = Number.MAX_VALUE;
```

Comment:
why MAX_VALUE?

Reply:
@chrischrischris Any large number is fine for changing the values of the parameters of the BudouX model. Should I replace it with another large number?

Reply:
I guess I don't understand why that's needed but it's fine if that's how it should be